QoL updates

chasen-bettinger opened the pull request:
- Upgraded CycloneDX to `0.19.0` from `0.14.0`
- Upgraded @cyclonedx/bom to `3.1.1`
- Added an `rm` call to `node-cdx-bom` to resolve bug that incurred
on multiple calls to the same repository
- Upgrade Dockerfile to use node16 and npm instead of yarn

Commit:
QoL updates

- Upgraded CycloneDX to `0.19.0` from `0.14.0`
- Upgraded @cyclonedx/bom to `3.1.1`
- Added an `rm` call to `node-cdx-bom` to resolve bug that incurred
on multiple calls to the same repository
- Upgrade Dockerfile to use node16 and npm instead of yarn

Commit:
v0.2.0

Commit:
Remove unnecessary echo statement

Commit:
Add info on how to build docker image

Commit:
Add updated yarn.lock with new package version

austinkelleher commented on Dockerfile:
```diff
@@ -1,11 +1,21 @@
-FROM node:12
+# GENERIC DOCKER SETUP
+FROM node:16
```
Fwiw, I don't think most engineering teams have adopted Node 16 because it isn't yet supported in Lambda. Our Jenkins builder agents are configurable to use Node 12 or Node 14 currently. Not sure if this matters for your use cases, but thought I would mention.

austinkelleher commented on Dockerfile:
```diff
@@ -1,11 +1,21 @@
-RUN yarn install --production
+
+# NPM PACKAGES
 RUN npm install -g @cyclonedx/bom
```
I realize this is old code, but wondering if we should install a specific version/range globally instead of the _latest_ by default? Seems dangerous to always install the latest. The package may introduce breaking changes across versions.

Something like:

```
RUN npm install -g @cyclonedx/bom@^3
```

Commit:
Updates based on PR feedback

- Use node 14
- Specify explicit version (3.1.1) for cyclonedx/bom NPM global
package

Commit:
Dockerfile NPM install consistent w/ package.json

erichs commented on Dockerfile:
```diff
@@ -1,11 +1,21 @@
-RUN npm install -g @cyclonedx/bom
+
+# NPM PACKAGES
+RUN npm install -g @cyclonedx/bom@^3.1.1
```
yay for reproducibility. good call!

erichs commented on bin/node-cdx-bom:
```diff
@@ -34,10 +34,18 @@ main() {
+  # to fail.
+  echo
+  echo "Removing pre-existing BOM file if there is one"
+  rm -f $output_bomfile
```
TIL that `-f` does not result in ENOENT or exit code 1 if file not found. Nifty.

erichs reviewed (dismissed):
Thanks!

chasen-bettinger committed:
Update CHANGELOG.md to be accurate with node14

Commit:
Remove yarn.lock; add package-lock.json
